tmt/steps/discover/fmf.py

```diff
@@ -341,6 +343,30 @@ def assert_git_url(plan_name: Optional[str] = None) -> None:
                 if not self.opt('dry'):
                     shutil.copytree(directory, self.testdir, symlinks=True)
 
+        # Prepare path of the dynamic reference file either following
+        # special syntax ref: @filepath or using the default location
+        if ref and ref.startswith("@"):
+            ref_filepath = os.path.join(self.testdir, ref[1:])
+            if not os.path.exists(ref_filepath):
+                raise tmt.utils.DiscoverError(
+                    f"Dynamic 'ref' definition file '{ref_filepath}' does not exist.")
+        else:
+            ref_filepath = os.path.join(self.testdir, DEFAULT_DYNAMIC_REF_FILEPATH)
+
+        # Apply dynamic referencing if the definition file exists
+        if os.path.exists(ref_filepath):
```

Comment:
This is not correct now. Previous condition
```
# dynamic ref check is enabled when ref is not defined or when it starts with @
if (not ref or str(ref)[0] == '@'):
    if os.path.exists(ref_filepath):
```
was there to make sure that we won't be loading `ref` from a default `.tmt/ref.fmf` file when `ref` is defined on a test plan level or on cmdline. The current code will process `.tmt/ref.fmf` despite the user specified specific branch in `ref` in a plan.

Reply:
I see, good catch!

```python
        if os.path.exists(ref_filepath) and (not ref or ref.startswith("@")):
```

The code above should be ok, right?

Reply:
Yes.. just please add the comment too since the reasoning may not be obvious from the condition itself.
To be fair, the former comment didn't make it obvious either. :-D

Reply:
Thanks, updated, includin the comment.